### ISSUE TITLE
Concerto Probing and Tuning issues

### ISSUE THREAD

**Issue by haomengz** [Pointcept/Pointcept#548]
Hi,

I have questions regarding the probing and tunning Concerto with the provided config files. Specifically, I follow #531 to use the [concerto_base_origin.pth](https://huggingface.co/Pointcept/Concerto/blob/main/concerto_base_origin.pth) and fully finetune the base model on ScanNet dataset with the command:
``
bash scripts/train.sh -m 1 -g 4 -d concerto -c semseg-ptv3-large-v1m1-0c-scannet-ft -n semseg-ptv3-large-v1m1-0c-scannet-ft -w exp/concerto/pretrain-concerto-v1m1-0-base-ckpt/concerto_base_origin.pth
``
The final results I obtain is: Val result: mIoU/mAcc/allAcc 0.7692/0.8436/0.9157, which is lower than (but comparable to) the base model performance reported in Table 3: 77.3/86.6/91.7. However, I am confused on the results in Table 2(a) where Concerto achieves 80.7/87.4/93.1 on ScanNet, as these numbers does not match any of the models in Table 3. Which model do you use for the results in Table 2(a), and how could I re-produce the same results?

Additionally, I try to reproduce the linear probing experiments on ScanNet with the command:
``
bash scripts/train.sh -m 1 -g 4 -d concerto -c semseg-ptv3-large-v1m1-0c-scannet-ft -n semseg-ptv3-large-v1m1-0c-scannet-ft -w exp/concerto/pretrain-concerto-v1m1-0-base-ckpt/concerto_base_origin.pth
``
as instructed in the [README file](https://github.com/haomengz/Pointcept/tree/main/pointcept/models/concerto#4-probing-and-tuning). I use the same pre-trained base model but only get Val result: mIoU/mAcc/allAcc 0.2025/0.2752/0.6480, which is significantly lower than the numbers in Table 2(b). Do I need to change to use the large model to reproduce the results?

**Comment by Yujia-Zhang0913** [comment]
Hi, do you mean that on ScanNet, your linear result is 0.2025/0.2752/0.6480, and the full fine-tuning result is 0.7692/0.8436/0.9157?
If so, I think the weight may not be loaded correctly. Do you follow the recently updated [README](https://github.com/Pointcept/Pointcept/tree/main/pointcept/models/concerto#4-probing-and-tuning) here to change the config to load the [concerto_base_origin.pth](https://huggingface.co/Pointcept/Concerto/blob/main/concerto_base_origin.pth)? Current config is for loading original locally pretrained weight, thus not suiting the concerto_small/base/large.pth on huggingface.

And you can first check your log file for linear probing: 
Does it contain the line below?
`Missing keys: ['module.seg_head.weight', 'module.seg_head.bias']`
The Missing keys should not contain any model part with `enc.`.

**Comment by haomengz** [comment]
Thanks for the reply. It turns out that I did not change the config which led to unreasonable results.

I am still confused on the results in Table 2(a) where Concerto achieves 80.7/87.4/93.1 on ScanNet, as these numbers does not match any of the models in Table 3. Which model do you use for the results in Table 2(a)? Why the numbers in Table 2(a) does not match any of the numbers in Table 3?

**Comment by Yujia-Zhang0913** [comment]
By full finetuning, Concerto achieves 80.7/87.4/93.1 on ScanNet in Table 2(a). The results of Table 3 are obtained by linear probing, where the base model results are the same as those in Table 2(b).

I use the base model for Table 2(a).

**Comment by haomengz** [comment]
Thanks for the clarification.

I’ve updated the training config and tried both linear probing and fine-tuning the base model on ScanNet. My results are:
Linear probing: 75.97 / 85.51 / 91.24
Fine-tuning: 80.00 / 88.76 / 92.83
(Using the checkpoint with the lowest validation loss.)

In the paper, the reported numbers are 77.3 / 86.6 / 91.7 and 80.7 / 87.4 / 93.1, respectively.
Is this level of discrepancy expected, or might I be missing something in the reproduction process?

**Comment by Yujia-Zhang0913** [comment]
Do you use the configs in the concerto folder with batch size 64 and learning rate 0.001 for fine-tuning, 0.002 for linear probing?
As in the https://github.com/Pointcept/Pointcept/issues/531, I have uploaded the corresponding configs to the concerto folder. It should not have problems reproducing.

**Comment by haomengz** [comment]
Yes, I use the same config files in the Concerto folder. Same batch size and learning rate. The only change is the checkpoint hook, as instructed in the [README](https://github.com/Pointcept/Pointcept/tree/main/pointcept/models/concerto#4-probing-and-tuning) file.
